### ISSUE TITLE
Saved model evaluate and validation always giving 0.50 accuracy despite reaching 96% on exact same data during training

### ISSUE THREAD

**Issue by aliencaocao** [tensorflow/tensorflow#53485]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): windows 10 21h2
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):  binary
- TensorFlow version (use command below):  v2.7.0-rc1-69-gc256c071bb2 2.7.0
- Python version: 3.9.9
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 11.3/8.2.1
- GPU model and memory: RTX 3080Ti 12GB

**Describe the current behavior**
When training a normal CNN model using the horse VS human dataset (binary image classification), the training accuracy reached 0.96+, but validation is forever stuck at 0.5. What's more interesting is after I load the saved model and called `model.evaluate` on the training set, it also gives 0.50, despite the training set being exactly same and was yielding 0.96+ during the actual training.

**Describe the expected behavior**
model to perform normally in both training and testing

model weight: https://drive.google.com/file/d/11oqaGvy4vV0v77WmWzQxhtQSgEH4PUhA/view?usp=sharing

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no): no
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
```
import tensorflow as tf
import urllib
import zipfile
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.layers import *


def download_data():
    _TRAIN_URL = "https://storage.googleapis.com/download.tensorflow.org/data/horse-or-human.zip"
    _TEST_URL = "https://storage.googleapis.com/download.tensorflow.org/data/validation-horse-or-human.zip"
    urllib.request.urlretrieve(_TRAIN_URL, 'horse-or-human.zip')
    local_zip = 'horse-or-human.zip'
    zip_ref = zipfile.ZipFile(local_zip, 'r')
    zip_ref.extractall('tmp/horse-or-human/')
    zip_ref.close()
    urllib.request.urlretrieve(_TEST_URL, 'testdata.zip')
    local_zip = 'testdata.zip'
    zip_ref = zipfile.ZipFile(local_zip, 'r')
    zip_ref.extractall('tmp/testdata/')
    zip_ref.close()


def solution_model():
    train_datagen = ImageDataGenerator(rescale=1. / 255,
                                       rotation_range=40,
                                       width_shift_range=0.2,
                                       height_shift_range=0.2,
                                       shear_range=0.2,
                                       zoom_range=0.2,
                                       horizontal_flip=True,
                                       fill_mode='nearest')
    test_datagen = ImageDataGenerator(rescale=1. / 255)

    train_generator = train_datagen.flow_from_directory(
        'tmp/horse-or-human',  # This is the source directory for training images
        target_size=(300, 300),
        batch_size=16,
        class_mode='binary')

    print('Validation data')
    validation_generator = test_datagen.flow_from_directory(
        'tmp/testdata',
        target_size=(300, 300),
        batch_size=32)

    opt = tf.keras.optimizers.Adam(learning_rate=1e-3, epsilon=1e-6)
    loss = tf.keras.losses.BinaryCrossentropy(from_logits=True)
    epoch = 20
    batch_size = 16
    callbacks = [
        tf.keras.callbacks.EarlyStopping(monitor='val_accuracy', min_delta=0, patience=5, verbose=1,
                                         mode='auto', baseline=None, restore_best_weights=True),
        tf.keras.callbacks.ReduceLROnPlateau(monitor='val_accuracy', factor=0.05, patience=5, verbose=1)
    ]

    xInput = Input([300, 300, 3])
    x = tf.cast(xInput, tf.float32)
    resnet50 = tf.keras.applications.efficientnet.EfficientNetB0(include_top=False, weights='imagenet')
    x = resnet50(x)
    x = Flatten()(x)
    x = Dropout(0.2)(x)
    x = Dense(256)(x)
    x = BatchNormalization(epsilon=1.001e-5)(x)
    x = Activation('relu')(x)
    xOutput = Dense(1)(x)  # from logits so no need activation
    model = tf.keras.models.Model(xInput, xOutput)

    model.compile(optimizer=opt, loss=loss, metrics='accuracy')
    model.summary()
    model.fit(train_generator, validation_data=validation_generator, steps_per_epoch=train_generator.samples // batch_size, epochs=epoch, callbacks=callbacks, verbose=1)
    return model



if __name__ == '__main__':
    # download_data()  # only need to run this once
    model = solution_model()
    model.save("mymodel.h5")
```

Evaluation code:

```
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.models import load_model

test_datagen = ImageDataGenerator(rescale=1. / 255)

print('Validation data')
validation_generator = test_datagen.flow_from_directory(
    'tmp/horse-or-human',
    target_size=(300, 300),
    batch_size=32,
    class_mode='categorical')

model = load_model('mymodel.h5')
model.summary()
model.evaluate(validation_generator)  # should be >0.96
```

**Other info / logs** Include any logs or source code that would be helpful to
`33/33 [==============================] - 13s 125ms/step - loss: 0.7544 - accuracy: 0.5000`

As you can see, loss value is actually normal (it started training with 2+ loss, now only 0.75)

**Comment by mohantym** [comment]
Hi @aliencaocao ! 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) 
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by aliencaocao** [comment]
Ok i have transferred it. Will close this now, thanks!
